src/libs/ReportUtils.js

```diff
@@ -419,12 +419,49 @@ function isPolicyExpenseChatAdmin(report, policies) {
     return policyRole === CONST.POLICY.ROLE.ADMIN;
 }
 
+/**
+ * Returns true if report has a parent and is therefore a Thread.
+ *
+ * @param {Object} report
+ * @returns {Boolean}
+ */
+function isThread(report) {
+    if (report && report.parentReportID && report.parentReportActionID) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Returns true if reportAction has a child.
+ *
+ * @param {Object} reportAction
+ * @returns {Boolean}
+ */
+function isThreadParent(reportAction) {
+    if (reportAction && reportAction.childReportID && reportAction.childReportID !== 0) {
+        return true;
+    }
+    return false;
+}
+
 /**
  * Get either the policyName or domainName the chat is tied to
  * @param {Object} report
  * @returns {String}
  */
 function getChatRoomSubtitle(report) {
+    if (isThread(report)) {
+        if (!getChatType(report)) {
+            return '';
+        }
+
+        // If thread is not from a DM or group chat, the subtitle will follow the pattern 'Workspace Name • #roomName'
+        const parentReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`];
+        const workspaceName = getPolicyName(parentReport);
+        const roomName = isChatRoom(parentReport) ? parentReport.displayName : '';
+        return roomName ? `${workspaceName} • ${roomName}` : `${workspaceName}`;
```

Comment:
this is returning an error while testing on dev.
Steps: 
⁦1. Create a workspace if you dont already have one
2. Create a room under this workspace. make it public. Let's call this the parent report.
3. Leave some comments in the parent report. Thread one of them. A new report will open. Let's call this the child report.
4. On another account, someone thats not a member of parent report. Paste the child report url. 
5. 💥 it breaks

It also completely blocks the UI
![image](https://github.com/Expensify/App/assets/14356145/32684b23-4438-4c24-aab6-79608875e67d)